ffi/derive/src/ffi_fn.rs

```diff
@@ -0,0 +1,223 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::Ident;
+
+use crate::impl_visitor::{ffi_output_arg, Arg, FnDescriptor};
+
+pub fn generate(fn_descriptor: &FnDescriptor) -> TokenStream {
```

Comment:
Consider adding `mod gen` and using it for namespacing.

Reply:
I don't understand

Reply:
What I meant to say was, 

There's a lot of functions with the `gen_` prefix. It might be more readable if they were all defined in a module named `gen`, so calls are `gen::docs` rather than `gen_docs`.